main.js

```diff
@@ -0,0 +1,165 @@
+const url ='http://localhost:5050/fibonacci/'
+const input = document.getElementById('input');
+const form = document.querySelector('.form-control');
+const urlResults = 'http://localhost:5050/getFibonacciResults';
+const btn = document.getElementById('btn');
+const result = document.getElementById('result');
+const resultsList = document.getElementById('resultsList');
+const messageError = document.getElementById('message');
+const checkBox = document.getElementById('checkbox');
+function fibonacci(inputNumber){
+    let fib = [0, 1];
+    let data = [];
+    
+    for(let i = 2; i <= inputNumber; i++) {
+      fib[i] = fib[i - 1] + fib[i - 2]; 
+      data.push(fib[i]);
+    }
+    
+    return data[data.length-1];
+      
+}
+function message (messageInput){
+  offSpinner();
+      messageError.classList.remove('d-none');
+      messageError.innerHTML=messageInput;
+      form.classList.remove('text-dark');
+      form.classList.add('text-danger');
+}
+function onSpinner(){
+  let spinnerOn = document.getElementById('spinner');
+  spinnerOn.classList.remove('d-none');
+  result.classList.add('d-none');
+}
+function offSpinner(){
+  const spinnerOff = document.getElementById('spinner');
+  spinnerOff.classList.add('d-none');
+}
+
+function onSpinnerResults(){
+    let spinnerOnResults = document.getElementById('spinnerResults');
+    spinnerOnResults.classList.remove('d-none');
+    resultsList.classList.add('d-none');
+}
+function offSpinnerResults(){
+    const spinnerOffResults = document.getElementById('spinnerResults');
+    spinnerOffResults.classList.add('d-none');
+}
+const sortBy = document.getElementById('sortBy');
+ const numAsc = document.getElementById('numAsc');
+ const numDesc = document.getElementById('numDesc');
+ const dateAsc = document.getElementById('dateAsc');
+ const dateDesc = document.getElementById('dateDesc');
+ const sort = document.getElementById('sort');
+ function toSort(sorted){
+  onSpinnerResults();
+    fetch(urlResults)
+     .then((response) =>{
+        return response.json()
+     })
+     .then(function(data){
+    
+    
+    let res = "";
+        const slicedArray= data.results.sort(sorted).slice(0,10);
+        for(let i=0; i<slicedArray.length;i++){
+          let date = new Date(data.results[i].createdDate);
+            res += "<li>"+'The Fibonacci of '+"<strong>"+slicedArray[i].number+"</strong>"+ ' is '+"<strong>"+slicedArray[i].result+"</strong>"+'. Calculated at: '+date+"</li>";
+            
+        }
+        offSpinnerResults();
+        resultsList.classList.remove('d-none');
+        resultsList.innerHTML = res;
+ }
+  
+ )
+ }

```

Comment:
not a good name and it should be two different function one for sort and one for display